plugins/woocommerce/src/Internal/StockNotifications/Notification.php

```diff
@@ -449,4 +458,29 @@ public function get_product_name() {
 
 		return $product->get_parent_id() ? $product->get_name() : $product->get_title();
 	}
+
+	public function get_email_link( $action ) {
+		$key = wp_generate_password( 20, false );
+		$hashed = wp_fast_hash( $key );
+
+		$this->set_email_link_action_key( $hashed );
+		$this->save();
+
+		return add_query_arg(
+			array(
+				'action' => $action,
+				'key'    => $key,
+				'id'     => $this->get_id(),
+			),
+			get_site_url()
+		);
+	}
```

Comment:
We will use this method to get links for the emails we send.  It will overwrite any existing key, because we will only send a given user one link per action, and a user will only be able to complete one action at a time: either 'verify' or 'unsubscribe' but never both at the same time.